modules/oe_translation_corporate_workflow/tests/src/Functional/CorporateWorkflowTranslationTest.php

```diff
@@ -0,0 +1,677 @@
+    $node = $node_storage->load($node->id());
+    // The new (current) published revision should have the new FR translation.
+    $this->assertEquals('My node 2 FR', $node->getTranslation('fr')->label());
+
+    // The previous published revisions have the old FR translation.
+    $revision_ids = $node_storage->revisionIds($node);
+    /** @var \Drupal\node\NodeInterface[] $revisions */
+    $revisions = $node_storage->loadMultipleRevisions($revision_ids);
+    foreach ($revisions as $revision) {
+      if ($revision->isPublished() && (int) $revision->get('version')->major === 1) {
+        $this->assertEquals('My node FR', $revision->getTranslation('fr')->label());
+        break;
+      }
+    }
+
+    // Create a new draft and validate it.
+    $node = $node_storage->load($node->id());
+    $node->set('title', 'My node 3');
+    $node->set('moderation_state', 'draft');
+    $node->save();
+    $this->moderateNode($node, 'validated');
+    $revision_ids = $node_storage->revisionIds($node);
+    // We have 4 new revisions now.
+    $this->assertCount(14, $revision_ids);
+    // Test the UI changes when we have a published version and we make a new
+    // validated one which increases the major version and which means we can
+    // now translate both the published version and the new validated major.
+    $this->drupalGet(Url::fromRoute('entity.node.local_translation', ['node' => $node->id()]));
+    // Assert we see a message telling the user what's what.
+    $this->assertSession()->pageTextContains('Your content has a new version (3.0.0) ahead of the currently published one (2.0.0). This means you can perform translations on both.');
+    $this->assertSession()->pageTextContains('Be aware that any translations you change or add to the published version will not move upstream to the new version.');
+
+    // Assert the published and validated revision sitution.
+    $published = $node_storage->load($node->id());
+    $this->assertEquals('published', $published->get('moderation_state')->value);
+    $this->assertEquals('2.0.0', $this->getEntityVersion($published));
+    $validated = $node_storage->loadRevision($node_storage->getLatestRevisionId($node->id()));
+    $this->assertEquals('validated', $validated->get('moderation_state')->value);
+    $this->assertEquals('3.0.0', $this->getEntityVersion($validated));
+
+    // Create a new translation for each and save them both as drafts.
+    $create_link_published = $this->getSession()->getPage()->find('css', 'tr[hreflang="fr"] td[data-version="2.0.0"] a');
+    $this->assertEquals('New translation', $create_link_published->getText());
+    $create_link_published->click();
+    $this->assertSession()->elementContains('css', '#edit-title0value-translation', 'My node 2 FR');
+    $values = [
+      'Translation' => 'My node 2 FR (updated but no actual change cause we already have a translation)',
+    ];
+    $this->submitForm($values, t('Save as draft'));
```

Comment:
Not sure why always using translatable string for the button in everywhere in the test. Backend form won't be translatable anyway, no?
```suggestion
    $this->submitForm($values, 'Save as draft');
```

Reply:
Boh, it's like that I think everywhere...so not gonna bother to remove the t wrapper now..